File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/recipes/RecipeWithEventTransformation.kt
package examples.kotlin.recipes

import com.ing.baker.recipe.kotlindsl.ExperimentalDsl
import com.ing.baker.recipe.kotlindsl.recipe
import examples.kotlin.events.OrderPlaced
import examples.kotlin.interactions.ShipOrder

@ExperimentalDsl
object RecipeWithEventTransformation {
    val recipe = recipe("example") {
        interaction<ShipOrder> {
           transformEvent<OrderPlaced>(newName = "OrderCreated") {
               "customerId" to "userId"
               "productIds" to "skus"
           }
        }
    }
}

File: examples/docs-code-snippets/src/main/scala/examples/scala/recipes/RecipeWithEventTransformation.scala
package examples.scala.recipes

import com.ing.baker.recipe.scaladsl.Recipe
import examples.scala.events.PaymentReceived
import examples.scala.interactions.ShipOrder

object RecipeWithEventTransformation {
  val recipe: Recipe = Recipe("example")
    .withInteraction(
      ShipOrder.interaction
        .withEventOutputTransformer(
          event = PaymentReceived.event,
          newEventName = "OrderCreated",
          ingredientRenames = Map.apply(
            ("customerId", "userId"),
            ("productIds", "skus")
          )
        )
    )
}